package com.xishuai.demo.springbootlog4j2;

import lombok.extern.log4j.Log4j2;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@Log4j2
@RestController
public class HelloController {

    @RequestMapping("/home")
    public String home() {

        log.error("Hi ! We have an Error. Hello World");

        return "Hello World ----spring-boot-log4j2";
    }

    @RequestMapping("/home2")
    public String home2() {

        //log.error("{\"level\":\"ERROR\",\"msg\":\"Failed to start connector [Connector[HTTP/1.1-8001]]\",\"createTime\":1526871267253,\"stackTrace\":\"org.apache.catalina.LifecycleException: Failed to start component [Connector[HTTP/1.1-8001]]\\n\\tat org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:167)\\n\\tat org.apache.catalina.core.StandardService.addConnector(StandardService.java:225)\\n\\tat org.springframework.boot.context.embedded.tomcat.TomcatEmbeddedServletContainer.addPreviouslyRemovedConnectors(TomcatEmbeddedServletContainer.java:250)\\n\\tat org.springframework.boot.context.embedded.tomcat.TomcatEmbeddedServletContainer.start(TomcatEmbeddedServletContainer.java:193)\\n\\tat org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.startEmbeddedServletContainer(EmbeddedWebApplicationContext.java:297)\\n\\tat org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.finishRefresh(EmbeddedWebApplicationContext.java:145)\\n\\tat org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:546)\\n\\tat org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122)\\n\\tat org.springframework.boot.SpringApplication.refresh(SpringApplication.java:693)\\n\\tat org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:360)\\n\\tat org.springframework.boot.SpringApplication.run(SpringApplication.java:303)\\n\\tat org.springframework.boot.SpringApplication.run(SpringApplication.java:1118)\\n\\tat org.springframework.boot.SpringApplication.run(SpringApplication.java:1107)\\n\\tat com.example.log4j2_demo.Log4j2DemoApplication.main(Log4j2DemoApplication.java:10)\\nCaused by: org.apache.catalina.LifecycleException: Protocol handler start failed\\n\\tat org.apache.catalina.connector.Connector.startInternal(Connector.java:1021)\\n\\tat org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)\\n\\t... 13 more\\nCaused by: java.net.BindException: Address already in use\\n\\tat sun.nio.ch.Net.bind0(Native Method)\\n\\tat sun.nio.ch.Net.bind(Net.java:433)\\n\\tat sun.nio.ch.Net.bind(Net.java:425)\\n\\tat sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:223)\\n\\tat sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:74)\\n\\tat org.apache.tomcat.util.net.NioEndpoint.bind(NioEndpoint.java:210)\\n\\tat org.apache.tomcat.util.net.AbstractEndpoint.start(AbstractEndpoint.java:1150)\\n\\tat org.apache.coyote.AbstractProtocol.start(AbstractProtocol.java:591)\\n\\tat org.apache.catalina.connector.Connector.startInternal(Connector.java:1018)\\n\\t... 14 more\\n\"}");
        //log.error("{\"msg\":\"出现一个异常错误：请求连接失败\",\"level\":\"ERROR\",\"createTime\":\"2018-5-21 20:22:22\",\"provider\":\"xishuai\",\"ip\":\"192.168.1.11\",\"stackTrace\":\"java.lang.Exception\\n\\tat com.example.log_demo.LogDemoTests.logCustomField(LogDemoTests.java:33)\\n\\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\\n\\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\\n\\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\\n\\tat java.lang.reflect.Method.invoke(Method.java:498)\\n\\tat org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)\\n\\tat org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)\\n\\tat org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)\\n\\tat org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)\\n\\tat org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:75)\\n\\tat org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:86)\\n\\tat org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:84)\\n\\tat org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)\\n\\tat org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:252)\\n\\tat org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:94)\\n\\tat org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)\\n\\tat org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)\\n\\tat org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)\\n\\tat org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)\\n\\tat org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)\\n\\tat org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)\\n\\tat org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70)\\n\\tat org.junit.runners.ParentRunner.run(ParentRunner.java:363)\\n\\tat org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:191)\\n\\tat org.junit.runner.JUnitCore.run(JUnitCore.java:137)\\n\\tat com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)\\n\\tat com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)\\n\\tat com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)\\n\\tat com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)\\n\",\"tag\":\"\",\"url\":\"\"}");
        log.error("{\"msg\":\"出现一个异常错误：请求连接失败\",\"level\":\"ERROR\",\"createTime\":\"2018-5-21 20:22:22\",\"provider\":\"xishuai\",\"ip\":\"192.168.1.11\",\"stackTrace\":\"java.lang.Exception\\n\\tat com.example.log_demo.LogDemoTests.logCustomField(LogDemoTests.java:33)\\n\\tat org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:191)\\n\\tat org.junit.runner.JUnitCore.run(JUnitCore.java:137)\\n\\tat com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)\\n\\tat com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)\\n\\tat com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)\\n\\tat com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)\\n\",\"tag\":\"\",\"url\":\"\"}");

        return "Hello World ----spring-boot-log4j2";
    }
}
